src/qibolab/platforms/platform.py

```diff
@@ -85,3 +124,76 @@ def set_bias(self, qubit, bias):
 
     def get_bias(self, qubit):
         return self.qubits[qubit].flux.bias
+
+
+class AcquisitionType(Enum):
+    """
+    Types of data acquisition from hardware.
+
+    SPECTROSCOPY: Zurich Integration mode for RO frequency sweeps,
+    INTEGRATION: Demodulate and integrate the waveform,
+    RAW: Acquire the waveform as it is,
+    DISCRIMINATION: Demodulate, integrate the waveform and discriminate among states based on the voltages
+
+    """
+
+    RAW = auto()
+    INTEGRATION = auto()
+    DISCRIMINATION = auto()
+
+
+class AveragingMode(Enum):
+    """
+    Types of data averaging from hardware.
+
+    CYLIC: Better averaging for noise,
+    SINGLESHOT: False averaging,
+    [SEQUENTIAL: Worse averaging for noise]
+
+    """
+
+    CYCLIC = auto()
+    SINGLESHOT = auto()
+
+
+@dataclass(frozen=True)
+class ExecutionParameters:
+    """Data structure to deal with execution parameters
+
+    :nshots: nshots (int): Number of shots to sample from the experiment. Default is 1024.
+    relaxation_time (int): Time to wait for the qubit to relax to its ground state between shots in s.
+                If ``None`` the default value provided as ``relaxation_time`` in the runcard will be used.
+    :fast_reset (bool): Enable or disable fast reset
+    :acquisition_type (AcquisitionType): Data acquisition mode
+    :averaging_mode (AveragingMode): Data averaging mode
+    """
+
+    nshots: Optional[int] = 1024
+    relaxation_time: Optional[float] = None
+    fast_reset: bool = False
+    acquisition_type: AcquisitionType = AcquisitionType.DISCRIMINATION
+    averaging_mode: AveragingMode = AveragingMode.SINGLESHOT
+
+    def __post_init__(self):
+        if not isinstance(self.acquisition_type, AcquisitionType):
+            raise TypeError("acquisition_type is not valid")
+        if not isinstance(self.averaging_mode, AveragingMode):
+            raise TypeError("averaging mode is not valid")
+
+    @property
+    def results_type(self):
+        """Returns corresponding results class"""
+        return RESULTS_TYPE[self.averaging_mode][self.acquisition_type]
+
+    RESULTS_TYPE = {
+        AveragingMode.CYCLIC: {
+            AcquisitionType.INTEGRATION: AveragedIntegratedResults,
+            AcquisitionType.RAW: AveragedRawWaveformResults,
+            AcquisitionType.DISCRIMINATION: AveragedStateResults,
+        },
+        AveragingMode.SINGLESHOT: {
+            AcquisitionType.INTEGRATION: IntegratedResults,
+            AcquisitionType.RAW: RawWaveformResults,
+            AcquisitionType.DISCRIMINATION: StateResults,
+        },
+    }
```

Comment:
This will not work because `RESULTS_TYPE` is a class variable. You should either change to
```py
return self.RESULTS_TYPE[...][...]
```
or move `RESULTS_TYPE` at the module level (outside the class scope). I would do the latter.

I would also move all these classes above the `DesignPlatform` class because they are used by the class and typically in Python you first define and then use. Even better we could move them to a seperate file and import them here.